src/device.hpp

```diff
@@ -313,6 +332,15 @@ struct cvk_device : public _cl_device_id,
 
     bool supports_subgroups() const { return m_has_subgroups_support; }
 
+    bool supports_controled_subgroups() const {
+        return (m_properties.apiVersion >= VK_MAKE_VERSION(1, 3, 0) ||
+                is_vulkan_extension_enabled(
+                    VK_EXT_SUBGROUP_SIZE_CONTROL_EXTENSION_NAME)) &&
```

Comment:
Maybe you could set a boolean like the other `m_has_*` to avoid repeating the conditions. Maybe we should make this automatic at some point.

Reply:
I am not sure to understand what condition we are repeating.

Reply:
The check for Vulkan 1.3 or the extension.